wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/aimodel/AiModelDataHandler.java

```diff
@@ -0,0 +1,232 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import gov.nist.itl.ssd.wipp.backend.core.CoreConfig;
+import gov.nist.itl.ssd.wipp.backend.core.model.data.DataHandler;
+import gov.nist.itl.ssd.wipp.backend.core.model.job.Job;
+import gov.nist.itl.ssd.wipp.backend.core.model.job.JobExecutionException;
+
+/**
+ * @author Mohamed Ouladi <mohamed.ouladi at nist.gov>
+ * @author Mylene Simon <mylene.simon at nist.gov>
+ * @author Antoine Meyer <antoine.meyer at nist.gov>
+ */
+@Component("aiModelDataHandler")
+@Qualifier("tensorflowModelDataHandler")
+public class AiModelDataHandler extends BaseDataHandler implements DataHandler {
+
+	@Autowired
+	CoreConfig config;
+
+    @Autowired
+    private AiModelRepository aiModelRepository;
+
+    @Autowired
+    private PluginRepository wippPluginRepository;
+
+    @Autowired
+    private TensorBoardLogsController tensorBoardLogsController;
+
+    @Autowired
+    private AiModelCardRepository modelCardRepository;
+
+	@Override
+	public void importData(Job job, String outputName) throws JobExecutionException {
+        // Get plugin
+        Plugin plugin = wippPluginRepository.findById(job.getWippExecutable()).orElse(null);
+        assert plugin != null;
+
+        AiModel aiModel = new AiModel(job, outputName);
+        // Set owner to job owner
+        aiModel.setOwner(job.getOwner());
+        // Set TM to private
+        aiModel.setPubliclyShared(false);
+        // Set framework
+        setFramework(plugin, aiModel);
+        aiModelRepository.save(aiModel);
+
+        File trainedModelFolder = new File(config.getAiModelsFolder(), aiModel.getId());
+        trainedModelFolder.mkdirs();
+
+        File tempOutputDir = getJobOutputTempFolder(job.getId(), outputName);
+        boolean success = tempOutputDir.renameTo(trainedModelFolder);
+        if (!success) {
+            aiModelRepository.delete(aiModel);
+            throw new JobExecutionException("Cannot move ai model to final destination.");
+        }
+        setOutputId(job, outputName, aiModel.getId());
+
+        // Create & save Model Card
+        AiModelCard mc = new AiModelCard(aiModel, job, plugin);
+        // Fill with TensorboardLogs data
+        try {
+            // Declare id
+            String id = job.getId();
+            // Declare variables
+            List<List<String>> data;
+            Float startTime, endTime, time, epochs, maxAccuracy, minLoss;
+            for(String type : new String[]{"train", "test"})
+            {
+                setAccuracy(type, id, mc);
+                setLoss(type, id, mc);
+            }
+        } catch (IOException e) { throw new RuntimeException(e); }
+
+        // Save
+        modelCardRepository.save(mc);
+	}
+
+    private void setFramework(Plugin plugin, AiModel aiModel) {
+        // search for output where "name" == "outputDir"
```

Comment:
Instead we should search for output that is of type `AiModel`, the name of the output could be different from `outputDir`